### PR TITLE
test

### PR DESCRIPTION
**Pull request by BenBeard115**

### CODE REVIEW

**Review by anukrr — DISMISSED** [review]

**Review by clv1 — DISMISSED** [review]

**Review by clv1 — APPROVED** [review]

File: Pipeline/example_extract.py [hunk 1]
```diff
@@ -1,17 +1,73 @@
...
-from sqlalchemy import create_engine, sql
+import requests
+import pandas as pd
+
+
+def get_plant_data() -> list[list[str]]:
+    plant_range = [num for num in range(0, 51
+                                        )]
+    url = "https://data-eng-plants-api.herokuapp.com/plants/"
+    # return [requests.get(f'{url}{plant}').json() for plant in plant_range]
```
Comment (ChrisOwen101) [inline Pipeline/example_extract.py:9]: Remove commented out code

File: Pipeline/example_extract.py [hunk 2]
```diff
@@ -1,17 +1,73 @@
...
-from dotenv import load_dotenv
-from sqlalchemy import create_engine, sql
+import requests
+import pandas as pd
+
+
+def get_plant_data() -> list[list[str]]:
+    plant_range = [num for num in range(0, 51
+                                        )]
+    url = "https://data-eng-plants-api.herokuapp.com/plants/"
```
Comment (ChrisOwen101) [inline Pipeline/example_extract.py:8]: Should this be a global variable?

File: Pipeline/example_extract.py [hunk 3]
```diff
@@ -1,17 +1,73 @@
...
+    # return [requests.get(f'{url}{plant}').json() for plant in plant_range]
+    data = []
+    i = 1
+    for plant in plant_range:
+        print(i)
+        i += 1
```
Comment (ChrisOwen101) [inline Pipeline/example_extract.py:14]: Why not use enumerate?

File: Pipeline/example_extract.py [hunk 4]
```diff
@@ -1,17 +1,73 @@
...
+
+        image_url = get_image(response)
+
+        longitude, latitude, town, country, country_abbreviation, continent = get_location(
+            response)
+
+        if response.get("scientific_name"):
+            scientific_name = response.get("scientific_name")[0]
+
+        data.append([response.get("plant_id"), response.get(
```
Comment (ChrisOwen101) [inline Pipeline/example_extract.py:30]: Is a list the best data structure for this?

File: Pipeline/example_extract.py [hunk 5]
```diff
@@ -1,17 +1,73 @@
...
+            "name"), scientific_name, response.get("last_watered"), response.get("recording_taken"), response.get("soil_moisture"), response.get("temperature"), email, first_name, last_name, phone_number, image_url, longitude, latitude, town, country, country_abbreviation, continent])
+    return data
+
+
+def get_image(response):
```
Comment (ChrisOwen101) [inline Pipeline/example_extract.py:35]: Type hints

File: Pipeline/example_extract.py [hunk 6]
```diff
@@ -1,17 +1,73 @@
...
+    if response.get("images"):
+        return response.get("images").get("original_url")
+
+
+def get_location(response: dict) -> list[str]:
```
Comment (ChrisOwen101) [inline Pipeline/example_extract.py:40]: What does this function return if location isn't found? Also the return type is wrong

File: Pipeline/example_extract.py [hunk 7]
```diff
@@ -1,17 +1,73 @@
...
+        longitude = location[0]
+        latitude = location[1]
+        town = location[2]
+        country_abbreviation = location[3]
+        country = location[4].split("/")[1]
+        continent = location[4].split("/")[0]
+    return longitude, latitude, town, country, country_abbreviation, continent
+
+
+def get_botanist(response: dict) -> list[str]:
```
Comment (ChrisOwen101) [inline Pipeline/example_extract.py:52]: Same as above

**Pull request merged**